app/javascript/components/buyers/BuyerProfile.jsx

```diff
@@ -8,20 +8,52 @@ class BuyerProfile extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      buyer: this.props.buyer
+      buyer: [],
+      canEditProfile: false,
+      componentDidMount: false
     }
   }
 
+  componentDidMount = () => {
+    const { user, userType, buyer } = this.props;
+    const buyer_id = this.props.buyer.id;
+    const buyer_route = APIRoutes.buyers.show(buyer_id);
+    Promise.all([
+      Requester.get(buyer_route)
+    ]).then(response => {
+      const [buyer_response] = response;
+      console.log(buyer_response);
+      this.setState({
+        buyer: buyer_response,
+        canEditProfile: userType === "buyer" && user.id === buyer.id,
```

Comment:
until cancancan is added, this errors out when nobody is logged in. can we add a `&& user` before the id check?